Sources/MultipartKit/MultipartPart.swift

```diff
@@ -24,9 +24,171 @@ public struct MultipartPart<Body: MultipartPartBodyElement>: Sendable {
-        get { self.headerFields.getParameter(.contentDisposition, "name") }
-        set { self.headerFields.setParameter(.contentDisposition, "name", to: newValue, defaultValue: "form-data") }
+    /// Parses and returns the Content-Disposition information from the part's headers.
+    ///
+    /// - Throws: `ContentDisposition.Error` if the header has an invalid format, or is missing required fields.
+    /// - Returns: A parsed `ContentDisposition` instance.
+    public var contentDisposition: ContentDisposition? {
+        get throws(ContentDisposition.Error) {
+            guard let field = self.headerFields[.contentDisposition] else {
+                return nil
+            }
+            return try .init(from: field)
+        }
+    }
+}
+
+/// Represents a parsed Content-Disposition header field for multipart messages.
+///
+/// The Content-Disposition header is defined in RFC 6266 (HTTP) and RFC 7578 (multipart/form-data)
+/// and provides metadata about each part, including:
+/// - The disposition type (form-data, attachment, inline)
+/// - The "name" parameter that identifies the form field (required for form-data)
+/// - An optional "filename" parameter for file uploads
+/// - Any additional custom parameters
+public struct ContentDisposition: Sendable {
+    /// The original header field value.
+    var underlyingField: String
+
+    /// The type of content disposition, indicating how the content should be handled.
+    ///
+    /// The disposition type provides a hint to the recipient about how the content
+    /// should be presented or processed.
+    public let dispositionType: DispositionType
+
+    /// The name parameter of the Content-Disposition header.
+    ///
+    /// This is a required parameter for multipart/form-data and represents
+    /// the name of the form field associated with this part.
+    public var name: String?
+
+    /// The optional filename parameter of the Content-Disposition header.
+    ///
+    /// When present, this indicates the part contains an uploaded file and
+    /// provides the original filename from the client.
+    public let filename: String?
+
+    /// Additional parameters included in the Content-Disposition header.
+    ///
+    /// These are any parameters beyond the standard "name" and "filename" parameters.
+    public let additionalParameters: [String: String]
+
+    /// Initializes a ContentDisposition by parsing a raw header field value.
+    ///
+    /// - Parameter field: The raw Content-Disposition header field value.
+    /// - Throws: `ContentDisposition.Error` if the header has an invalid format, contains an
+    ///           unrecognized disposition type, or is missing required fields.
+    public init(from field: HTTPFields.Value) throws(ContentDisposition.Error) {
+        self.underlyingField = field
+
+        var parameters =
+            field
+            .split(separator: ";")
+            .map { $0.trimmingCharacters(in: .whitespaces) }
```

Comment:
Import `Algorithms` and:
```suggestion
            .map { $0.trimming(while: \.isWhitespace) }
```